### PR TITLE
golang repl and debugger

### PR DESCRIPTION
**Pull request by saolsen**
PR checklist:

- [ ] Added changelog entry.

### CODE REVIEW

File: languages/go/oso.go [hunk 1]
```diff
@@ -150,3 +150,7 @@ func (o Oso) IsAllowed(actor interface{}, action interface{}, resource interface
 		return false, nil
 	}
 }
+
+func (o Oso) Repl() error {
```
Comment (leina05) [inline languages/go/oso.go:154]: Can we add a comment above for API docs?

File: languages/go/polar.go [hunk 2]
```diff
@@ -135,7 +138,43 @@ func (p Polar) queryRule(name string, args ...interface{}) (*Query, error) {
...
+		}
+		if len(results) == 0 {
+			fmt.Println(false)
+		} else {
+			for _, bindings := range results {
+				for k, v := range bindings {
+					switch v := v.(type) {
+					// print strings with quotes but not variables or other types represented by strings
+					case string:
+						fmt.Printf("%v = %#v\n", k, v)
```
Comment (leina05) [inline languages/go/polar.go:170]: What does the `#` do?

File: languages/go/query.go [hunk 3]
```diff
@@ -94,8 +97,7 @@ func (q *Query) Next() (*map[string]interface{}, error) {
 			defer q.ffiQuery.Delete()
 			return nil, nil
 		case QueryEventDebug:
-			// TODO
-			return nil, fmt.Errorf("Polar debugger is not yet implemented in Go.")
+			err = q.handleDebug(ev)
```
Comment (leina05) [inline languages/go/query.go:100]: Do you need to return the error here?

**Review by leina05 — COMMENTED** [review]
Looks good, a few questions. Have to try out the debugger still.

- Do we need a test for the debugger at the Go level (not sure if we have this for other libs)?
- The REPL doesn't print anything for a simple "true" with no bindings. Do we want to print "true"? Not sure what the standard behavior is here

File: languages/go/polar.go (hunk 2 above)
Comment (saolsen) [reply languages/go/polar.go:170]: It prints the value as a go source literal, in this case that means adding quotes around it.

File: languages/go/query.go (hunk 3 above)
Comment (saolsen) [reply languages/go/query.go:100]: Ah yeah I think so.

File: languages/go/oso.go (hunk 1 above)
Comment (saolsen) [reply languages/go/oso.go:154]: Yeah

File: languages/go/query.go (hunk 3 above)
Comment (saolsen) [reply languages/go/query.go:100]: No, I don't it gets returned below.

**Review by leina05 — APPROVED** [review]

**Pull request merged**